test/parallel/test-http-perf_hooks.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http = require('http');
+
+const { PerformanceObserver } = require('perf_hooks');
+
+const obs = new PerformanceObserver(common.mustCall((items) => {
+  const entry = items.getEntries()[0];
+  assert.strictEqual(entry.entryType, 'http');
+  assert.strictEqual(typeof entry.startTime, 'number');
+  assert.strictEqual(typeof entry.duration, 'number');
+}, 1));
+
+// Should throw if entryTypes are not valid
+{
+  const expectedError = { code: 'ERR_VALID_PERFORMANCE_ENTRY_TYPE' };
+  const wrongEntryTypes = { entryTypes: ['foo', 'bar', 'baz'] };
+  assert.throws(() => obs.observe(wrongEntryTypes), expectedError);
+}
+
+obs.observe({ entryTypes: ['http'] });
+
+const expected = 'Post Body For Test';
+
+const server = http.Server(function(req, res) {
+  let result = '';
+
+  req.setEncoding('utf8');
+  req.on('data', function(chunk) {
+    result += chunk;
+  });
+
+  req.on('end', function() {
+    assert.strictEqual(result, expected);
+    server.close();
+    res.writeHead(200);
+    res.end('hello world\n');
+  });
+
+});
+
+server.listen(0, function() {
+  for (let i = 0; i < 1; i++) {
```

Comment:
Is this loop needed?

Reply:
Not in this case but i will modify this test to verify that we actually emit 2 entry when we receive 2 request.